helperFunctions/subscribeHelpers/SalesforceAgentAssignedHelpers/SubscribeToLiveAgentHelper.ts

```diff
@@ -69,24 +73,44 @@ export class SubscribeToLiveAgent {
 						if (chatEndedReason === 'agent') {
 							await updateRoomCustomFields(this.data.room.id, { agentEndedChat: true }, this.read, this.modify);
 							this.modify.getAnalytics().sendEvent(getEventData(this.data.room.id, EventName.CHAT_CLOSED_BY_AGENT));
+							await handleEndChatCallback.handleEndChat(CloseMessages.CLOSED_BY_AGENT);
+						} else {
+							await handleEndChatCallback.handleEndChat(CloseMessages.CLOSED_BY_VISITOR);
 						}
-						await handleEndChatCallback.handleEndChat();
+					} else if (isChatRequestFail === true) {
+						const {
+							message: { reason: chatEndedReason },
+						} = getForEvent(messageArray, 'ChatRequestFail');
+						if (chatEndedReason === 'Unavailable') {
+							await updateRoomCustomFields(this.data.room.id, { agentDisconnected: true }, this.read, this.modify);
+							this.modify.getAnalytics().sendEvent(getEventData(this.data.room.id, EventName.CHAT_CLOSED_LIVEAGENT_DISCONNECTED));
+							await handleEndChatCallback.handleEndChat(CloseMessages.LIVEAGENT_DISCONNECTED);
+						}
+						const event = getEventData(this.data.room.id, EventName.CHAT_CLOSED_UNKNOWN_ERROR_IN_CHECKING_AGENT_RESPONSE);
```

Comment:
Two events are being sent in this block. this one and the above one. I think we need only one?